dali/kernels/signal/moving_mean_square.cc

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/kernels/signal/moving_mean_square.h"
+
+namespace dali {
+namespace kernels {
+namespace signal {
+
+namespace {
+
+template<typename Result = float, typename T>
+float Square(const T &val) {
+  Result res = val;
+  return res * res;
+}
+
+
+template<typename T>
+float CalcSumSquared(span<const T> values) {
+  float sumsq = 0;
+  for (const auto &val : values) {
+    sumsq += Square(val);
+  }
+  return sumsq;
+}
+
+}  // namespace
+
+template<typename T>
+MovingMeanSquareCpu<T>::~MovingMeanSquareCpu() = default;
+
+
+template<typename T>
+KernelRequirements
+MovingMeanSquareCpu<T>::Setup(KernelContext &context, const InTensorCPU<T, 1> &in,
+                              const MovingMeanSquareArgs &args) {
+  DALI_ENFORCE(args.window_size < in.num_elements(),
+               make_string("window_size can't be bigger than input buffer. Obtained: window_size=",
```

Comment:
```suggestion
               make_string("window_size can't be bigger than input buffer. Got: window_size=",
```
It's not really "obtained" - it's been provided this way by the caller.

Reply:
Changed